homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,254 @@
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_SWING_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Daikin HVAC platform."""
+    customize = config.get(CONF_CUSTOMIZE, {})
+
+    devices = []
+
+    name = config.get(CONF_NAME, None)
+
+    if discovery_info is not None:
+        ip_address = discovery_info['ip']
+        devices.append(setup_hvac(ip_address, name, customize))
+    else:
+        ip_address = config.get(CONF_IP_ADDRESS, None)
+        devices.append(setup_hvac(ip_address, name, customize))
+
+    add_devices(devices)
+
+
+def setup_hvac(ip_address, name, customize):
+    if ip_address is None:
+        _LOGGER.error("Missing required configuration items %s",
+                      CONF_IP_ADDRESS)
+        return False
+
+    import pydaikin.appliance as appliance
+    device = appliance.Appliance(ip_address)
+
+    if name is None:
+        name = device.values['name']
+
+    return DaikinHVAC(device, name, customize)
+
+
+class DaikinHVAC(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, device, name, customize):
+        import pydaikin.appliance as appliance
+
+        """Initialize the HVAC."""
+        self._name = name
+        self._device = device
+        self._customize = customize
+
+        self._current_temperature = self.settings('htemp', True)
+        self._operation_list = list(DAIKIN_OPERATION_LIST.values())
+        self._fan_list = appliance.daikin_values('f_rate')
+        self._swing_list = appliance.daikin_values('f_dir')
+
+        self._target_temperature = None
+        self._current_humidity = None
+        self._target_humidity = None
+        self._current_fan_mode = None
+        self._current_swing_mode = None
+
+    @property
+    def unique_id(self):
+        """Return the ID of this AC."""
+        return "{}.{}".format(self.__class__, self._device.ip)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def should_poll(self):
+        """Polling needed for a demo climate device."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        self._current_temperature = self.settings('htemp', True)
+
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
+
+        return self._target_temperature
+
+    def set_temperature(self, **kwargs):
```

Comment:
`set_temperature` can also be called with an operation mode to set operation mode and temperature in one call, see [0]. This must be handled here.

[0] https://home-assistant.io/components/climate/#service-climateset_temperature

Reply:
I saw indeed the parameter "operation_mode" indeed but I would not do it here, because
1. it's an optional parameter
2. we have a dedicated method to setting the operation mode

I'll try to handle it later on tonight although I'm not sure I would want to make two api calls for updating two different values.
Implementation also depends on what the update method is supposed to do as I've asked earlier in another comment

Reply:
The problem of not handling the `operation_mode` in `set_temperature` is that it breaks scenes: Currently, scenes can only send one service call per entity. Thus, if a scene should set temperature *and* operation mode on a climate device, setting the operation mode must be implemented in the `set_temperature` method. That's not optimal, I agree, but it currently is a limitation.

Reply:
Let me see if I got this right: there is a known problem in scenes that should be threated in each climate component instead of trying to fix the root case in a single place and that place is in scenes :p ?

What if someone wants to set fan direction, target temperature, min temp, max temp and operation mode. That's not possible either in one call per entity from what u r saying but that doesn't mean we should handle it here, right ?

PS: As mentioned earlier I'll try to fix it later on tonight  ONLY IF I can avoid making two api calls (1 for temperature + 1 for operation mode) only for the love of the (partially) broken scenes.

Reply:
Found a way to achieve your suggestion without making two API calls. I'll submit the code shortly

Reply:
You're absolutely right: This is something that should be fixed in the scene component. However, that is probably something for a later PR. ;-)